src/custom-apps/components/PermissionAlert/utils.ts

```diff
@@ -0,0 +1,112 @@
+import { gql } from "@apollo/client";
+import { parse, visit } from "graphql";
+
+// cannot be in `queries.ts` as codegen cannot handle `__schema`
+export const IntrospectionQuery = gql`
+  query PermissionIntrospection {
+    __schema {
+      types {
+        kind
+        name
+        description
+        fields(includeDeprecated: false) {
+          name
+          description
+        }
+      }
+    }
+  }
+`;
+
+const uniq = (value: any, index: number, self: any) =>
+  self.indexOf(value) === index;
+
+const extractPermissions = (description: string) => {
+  const match = (description || "").match(/following permissions(.*): (.*?)\./);
```

Comment:
suggestion: If you expect that description may not exists should mark as optional/
```
const extractPermissions = (description?: string) => {
  const match = (description || "").match(/following permissions(.*): (.*?)\./);
```